recipes/hazelcast-cpp-client/all/test_package/CMakeLists.txt

```diff
@@ -1,9 +1,17 @@
 cmake_minimum_required(VERSION 3.10)
-project(PackageTest CXX)
+project(test_package CXX)
 
-include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup()
+if (${HAZELCAST_CPP_CLIENT_VERSION} VERSION_LESS_EQUAL "4.0.0")
+    find_package(hazelcastcxx REQUIRED CONFIG)
+else()
+    find_package(hazelcast-cpp-client REQUIRED CONFIG)
+endif()
 
 add_executable(test_package test_package.cpp)
-target_link_libraries(test_package PUBLIC ${CONAN_LIBS})
+if (${HAZELCAST_CPP_CLIENT_VERSION} VERSION_LESS_EQUAL "4.0.0")
+    target_link_libraries(test_package PUBLIC hazelcastcxx::hazelcastcxx)
```

Comment:
I recall this one since it was nightmare... Conan was not able to model it when I wrote it initially.

https://github.com/hazelcast/hazelcast-cpp-client/blob/v4.0.0/CMakeLists.txt#L257

https://github.com/hazelcast/hazelcast-cpp-client/blob/29482d8b096ef0e9a1883712434ac520a7f56380/CMakeLists.txt#L221

The files should get a `_STATIC` suffix and the targets change with the same namespace 🤔